packages/rum-core/src/domain/foregroundContexts.ts

```diff
@@ -148,6 +150,9 @@ function getInForegroundPeriods(eventStartTime: RelativeTime, duration: Duration
 
   for (let i = foregroundPeriods.length - 1; i >= 0; i--) {
     const foregroundPeriod = foregroundPeriods[i]
+    if (foregroundPeriods.length - i > MAX_NUMBER_OF_FOCUSED_TIME_PER_VIEW) {
+      break
+    }
```

Comment:
There is two stop conditions here. They may be merged together, with something like:

```suggestion
  const earliestIndex  = Math.max(0, foregroundPeriods.length - MAX_NUMBER_OF_FOCUSED_TIME_PER_VIEW)
  for (let i = foregroundPeriods.length - 1; i >= earliestIndex; i--) {
    const foregroundPeriod = foregroundPeriods[i]
```